ion_hash_test.go

```diff
@@ -0,0 +1,536 @@
+				testCase = append(testCase, byte(intValue))
+			}
+			err = reader.StepOut()
+			assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+
+		} else {
+			// Create textWriter to set testName to Ion text.
+			str := strings.Builder{}
+			textWriter := ion.NewTextWriter(&str)
+
+			// Create binaryWriter to set testCase to Ion binary.
+			buf := bytes.Buffer{}
+			binaryWriter := ion.NewBinaryWriter(&buf)
+
+			writeToWriter(t, reader, textWriter, binaryWriter)
+
+			err = textWriter.Finish()
+			assert.NoError(t, err, "Something went wrong writing Ion value to text writer.")
+			err = binaryWriter.Finish()
+			assert.NoError(t, err, "Something went wrong writing Ion value to binary writer.")
+
+			if testName == "unknown" {
+				testName = str.String()
+			}
+			if len(testCase) == 0 {
+				testCase = buf.Bytes()
+			}
+		}
+
+		// Iterate through expected/ digest bytes.
+		reader.Next()
+		fieldName := reader.FieldName()
+
+		if fieldName == "expect" {
+			err = reader.StepIn()
+			assert.NoError(t, err, "Something went wrong executing reader.StepIn()")
+
+			for reader.Next() {
+				identityUpdateList := [][]byte{}
+				identityDigestList := [][]byte{}
+				identityFinalDigest := [][]byte{}
+				md5UpdateList := [][]byte{}
+				md5DigestList := [][]byte{}
+
+				fieldName = reader.FieldName()
+				hasherName := fieldName
+				if fieldName == "identity" {
+					err = reader.StepIn()
+					assert.NoError(t, err, "Something went wrong executing reader.StepIn()")
+
+					for reader.Next() {
+						annotations := reader.Annotations()
+
+						if len(annotations) > 0 {
+							if annotations[0] == "update" {
+								updateBytes := readSexpAndAppendToList(t, reader)
+								identityUpdateList = append(identityUpdateList, updateBytes)
+
+							} else if annotations[0] == "digest" {
+								digestBytes := readSexpAndAppendToList(t, reader)
+								identityDigestList = append(identityDigestList, digestBytes)
+							} else if annotations[0] == "final_digest" {
+								digestBytes := readSexpAndAppendToList(t, reader)
+								identityFinalDigest = append(identityFinalDigest, digestBytes)
+							}
+						}
+					}
+					err = reader.StepOut()
+					assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+				} else if fieldName == "md5" {
+					err = reader.StepIn()
+					assert.NoError(t, err, "Something went wrong executing reader.StepIn()")
+
+					for reader.Next() {
+						annotations := reader.Annotations()
+
+						if len(annotations) > 0 {
+							if annotations[0] == "update" {
+								updateBytes := readSexpAndAppendToList(t, reader)
+								md5UpdateList = append(md5UpdateList, updateBytes)
+
+							} else if annotations[0] == "digest" {
+								digestBytes := readSexpAndAppendToList(t, reader)
+								md5DigestList = append(md5DigestList, digestBytes)
+							}
+						}
+					}
+					err = reader.StepOut()
+					assert.NoError(t, err, "Something went wrong executing reader.StepOut()")
+				}
```

Comment:
The only difference between the `fieldName == "identity"` case and `fieldName == "md5"` case is the "final_digest" logic. Why not just combine the two cases? Is there any harm in allowing the "md5" case to check for a "final_digest" annotation?

Reply:
Well we are updating different arrays. For MD5 we don't have md5FinalDigest list